gapis/atom/group_test.go

```diff
@@ -419,6 +419,107 @@ func TestAddAtomsSparse(t *testing.T) {
 	assert.With(ctx).That(got).DeepEquals(expected)
 }
 
+func TestAddAtomsWithSplitting(t *testing.T) {
+	ctx := log.Testing(t)
+	got := buildTestGroup(600)
+
+	got.AddAtoms(func(ID) bool { return true }, 45)
+
+	expected := Group{
+		"root", Range{0, 700}, Spans{
+			Group{"Sub Group 1", Range{0, 45}, Spans{Range{0, 45}}},
+			Group{"Sub Group 2", Range{45, 90}, Spans{Range{45, 90}}},
+			Group{"Sub Group 3", Range{90, 234}, Spans{
+				Range{90, 100},
+				Group{"Sub-group 0", Range{100, 200}, Spans{
+					Group{"Sub Group 1", Range{100, 145}, Spans{Range{100, 145}}},
+					Group{"Sub Group 2", Range{145, 190}, Spans{Range{145, 190}}},
+					Group{"Sub Group 3", Range{190, 200}, Spans{Range{190, 200}}},
+				}},
+				Range{200, 234},
+			}},
+			Group{"Sub Group 4", Range{234, 279}, Spans{Range{234, 279}}},
+			Group{"Sub Group 5", Range{279, 423}, Spans{
+				Range{279, 300},
+				Group{"Sub-group 1", Range{300, 400}, Spans{
+					Group{"Sub Group 1", Range{300, 364}, Spans{
+						Range{300, 340},
+						Group{"Sub-group 1.0", Range{340, 360}, Spans{
+							Range{340, 360},
+						}},
+						Range{360, 364},
+					}},
+					Group{"Sub Group 2", Range{364, 400}, Spans{Range{364, 400}}},
+				}},
+				Range{400, 423},
+			}},
+			Group{"Sub Group 6", Range{423, 468}, Spans{Range{423, 468}}},
+			Group{"Sub Group 7", Range{468, 612}, Spans{
+				Range{468, 500},
+				Group{"Sub-group 2", Range{500, 600}, Spans{
+					Group{"Sub Group 1", Range{500, 545}, Spans{Range{500, 545}}},
+					Group{"Sub Group 2", Range{545, 590}, Spans{Range{545, 590}}},
+					Group{"Sub Group 3", Range{590, 600}, Spans{Range{590, 600}}},
+				}},
+				Range{600, 612},
+			}},
+			Group{"Sub Group 8", Range{612, 657}, Spans{Range{612, 657}}},
+			Group{"Sub Group 9", Range{657, 700}, Spans{Range{657, 700}}},
+		},
+	}
+
+	assert.With(ctx).That(got).DeepEquals(expected)
+}
+
+func TestAddAtomsWithSplittingBoundry(t *testing.T) {
```

Comment:
Done. It started out as testing AddAtoms with evil boundary conditions, until I realized that AddAtoms collapses successive spans before the splitting code ever got into play and so it wasn't exercising the things I wanted it to...